tests/unit/test_utils.py

```diff
@@ -1263,3 +1264,21 @@ def test_deprecated_message_reads_well():
         "You can find discussion regarding this at "
         "https://github.com/pypa/pip/issues/100000."
     )
+
+
+@pytest.mark.parametrize("unbuffered", [False, True])
+def test_make_setuptools_shim_args(unbuffered):
+    built_shim = make_setuptools_shim_args("setup.py", unbuffered=unbuffered)
+    shim = (
+        "import sys, setuptools, tokenize;"
+        "sys.argv[0] = 'setup.py'; __file__='setup.py';"
+        "f=getattr(tokenize, 'open', open)(__file__);"
+        "code=f.read().replace('\\r\\n', '\\n');"
+        "f.close();"
+        "exec(compile(code, __file__, 'exec'))"
+    )
+    base_cmd = [sys.executable, "-c", shim]
+    if unbuffered:
+        base_cmd.insert(1, "-u")
```

Comment:
This test is basically just copying the implementation `make_setuptools_shim_args()`. Instead, to make the test more independent, I would just do some spot checks. For example, check that `-u` is in the list if and only if `unbuffered` is true. Check that `-c` is the second-last element. And check that `sys.argv[0] = 'setup.py'` is in the final argument. And I think that would be good enough.